components/eventing-controller/pkg/handlers/nats.go

```diff
@@ -125,8 +128,32 @@ func (n *Nats) SyncSubscription(sub *eventingv1alpha1.Subscription, cleaner even
 		cleanSubjects = append(cleanSubjects, subject)
 	}
 
+	// if subscription filters are modified, then delete subscriptions from NATS
+	// so that new subscriptions are created accordingly
+	if len(sub.Status.CleanEventTypes) != len(cleanSubjects) ||
+		!reflect.DeepEqual(sub.Status.CleanEventTypes, cleanSubjects) {
```

Comment:
do we care for order here? Imho we should not recreate the subscriptions just because the order changed

Reply:
It will be covered in the follow up [issue](https://github.com/kyma-project/kyma/issues/13108)